Implement box model test from 2001 JCP paper

mattdturner opened the pull request:
This PR implements the box model test defined in Hunke, E.C., 2001. Viscout-plastic sea ice dynamics with the EVP model: linearization issues. J. Comp. Phys. 170, 18-38

- Developer(s): Matt Turner, Rick Allard

- Please suggest code Pull Request reviewers in the column at right.

- Are the code changes bit for bit, different at roundoff level, or more substantial? bfb

- Is the documentation being updated with this PR? (Y/N) N
If not, does the documentation need to be updated separately at a later time? (Y/N) Y
Note: "Documentation" includes information on the wiki and .rst files in doc/source/, 
which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.

- Other Relevant Details:
- Majority of work performed by @rallard77 
- To run the test ``./cice.setup -m conrad --test smoke -s box2001 --testid t00 --grid gbox80``

mattdturner commented:
Unless I'm mistaken, modifications to ``ice_init.F90`` and ``ice_grid.F90`` will likely need to be made prior to merging.  Right now, it looks like some values were changed in order for this case to run (although I could be mistaken).

rallard77 commented:
All,

I modified dxrect, dyrect from 30.e5 to 16.e5 in ice_grid.F90. We can 
set grid spacing with a namelist variable.

This test case does not presently have constant coriolis. I used kdyn=1. 
The Hunke 2001 paper used kdyn=3 (not presently supported).

Rick

On 6/20/2018 12:59 PM, Matthew Turner wrote:
>
> Unless I'm mistaken, modifications to |ice_init.F90| and 
> |ice_grid.F90| will likely need to be made prior to merging. Right 
> now, it looks like some values were changed in order for this case to 
> run (although I could be mistaken).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-398841899>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h3UmBHy18MfJ4qoKpnEt0eCfd-YPks5t-o2RgaJpZM4Uvuf4>.
>

rallard77 commented:
The constant coriolis and namelist settings to define dx and dy still 
need to be set up.

Rick

On 6/24/2018 6:38 PM, Tony Craig wrote:
>
> *@apcraig* commented on this pull request.
>
> Are there things outstanding still to do, like setting the grid box 
> size or the coriolis setup? I agree it makes sense to add a couple 
> namelist to define dx and dy for the box grids. We will probably 
> need/want a few different grids for the various cases.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#pullrequestreview-131440737>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h7YVke_zN4bovfcbF2gX0jfP9LRoks5uACMHgaJpZM4Uvuf4>.
>

mattdturner commented:
I just pushed modifications that include a constant coriolis (included as a namelist option), and there are also namelist settings to define dx and dy.

mattdturner commented:
This PR has been updated to have dxrect and dyrect in the namelist instead of x_spacing and y_spacing.

Documentation has not yet been addressed

apcraig commented:
Is there a run that validates against the original results?

mattdturner commented:
@apcraig , not yet.  I run the full base_suite to make sure these changes didn't introduce new errors, but I can go ahead and generate a baseline and compare this PR to the baseline.  Do you think I should make the baseline from ``master`` or from ``cice6.0``?

rallard77 commented:
I don't have the original results from CICE4. Elizabeth, do you have any 
output files, say from a three day run so we can at least look at the 
results qualitatively?

Rick

On 8/2/2018 5:36 PM, Tony Craig wrote:
>
> Is there a run that validates against the original results?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-410089410>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h3jSaLSvFX7qpZw2lOU_VMlQ1QHrks5uM38KgaJpZM4Uvuf4>.
>

mattdturner commented:
I included set_nml.gbox80 so that users could use the gbox80 grid without needing to use the box2001 test.  So, users could use the gbox80 configuration to build their own test.

eclare108213 commented:
The code runs but isn't configured correctly.  There are several issues:

- It looks like block boundaries aren't being handled correctly.  I would guess that this is in the initialization.
- There should be land cells around the entire domain, not just the 2 corners.
- The 2001 test is for dynamics only, so ice area and thickness should remain constant.  That requires turning off the thermodynamics and the ridging.  We can create additional tests with those things turned on, once we get this basic one working.
- The ocean currents do not have the correct pattern (should be circular).  I'm not sure if the atmo vectors are correct, they are at least pointed in roughly the right direction but also seem to have the block-boundary problem.

![aice](https://user-images.githubusercontent.com/8173983/43660515-83a329f8-971c-11e8-9a02-9eff2612d6d1.png)
![uvel](https://user-images.githubusercontent.com/8173983/43660517-83ed7526-971c-11e8-8e0b-b97c1dbd0820.png)
![ocncurrent](https://user-images.githubusercontent.com/8173983/43660516-83d79990-971c-11e8-8a50-bbfedda0c216.png)
![wind](https://user-images.githubusercontent.com/8173983/43660519-8402fd6a-971c-11e8-8c2b-d796b6bd2437.png)

rallard77 commented:
Let me take a look....

Rick

On 8/3/2018 2:04 PM, Elizabeth Hunke wrote:
>
> The code runs but isn't configured correctly. There are several issues:
>
>   * It looks like block boundaries aren't being handled correctly. I
>     would guess that this is in the initialization.
>   * There should be land cells around the entire domain, not just the
>     2 corners.
>   * The 2001 test is for dynamics only, so ice area and thickness
>     should remain constant. That requires turning off the
>     thermodynamics and the ridging. We can create additional tests
>     with those things turned on, once we get this basic one working.
>   * The ocean currents do not have the correct pattern (should be
>     circular). I'm not sure if the atmo vectors are correct, they are
>     at least pointed in roughly the right direction but also seem to
>     have the block-boundary problem.
>
> aice 
> <https://user-images.githubusercontent.com/8173983/43660515-83a329f8-971c-11e8-9a02-9eff2612d6d1.png>
> uvel 
> <https://user-images.githubusercontent.com/8173983/43660517-83ed7526-971c-11e8-8e0b-b97c1dbd0820.png>
> ocncurrent 
> <https://user-images.githubusercontent.com/8173983/43660516-83d79990-971c-11e8-8a50-bbfedda0c216.png>
> wind 
> <https://user-images.githubusercontent.com/8173983/43660519-8402fd6a-971c-11e8-8c2b-d796b6bd2437.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-410347666>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h8cfETEoUEah0V3Yo0oryOGVNMCeks5uNJ7EgaJpZM4Uvuf4>.
>

rallard77 commented:
We changed the code to compile on only one processor and set nproc=1. The currents now look reasonable. See image below.
Set kdyn=0,  thickness does not change but aice is still changing. The "edeco" code specifes open boundaries for "ew_boundary_type" and "ns_boundary_type. Please advise. 
![currents_1x1](https://user-images.githubusercontent.com/29209991/43851187-9415da56-9aff-11e8-8be4-38f5bf6acebc.png)

eclare108213 commented:
kdyn=0 defeats the purpose of the 2001 test -- it turns the dynamics off and leaves the thermo on.  We need to turn the thermo off and have the dynamics on.  Turning the thermo off is a little complicated -- there is not a namelist flag to do that.  It will need to be added.  However, the wind stress needed for the dynamics is calculated as part of the thermodynamics.   In the past I've probably handled that by specifying the wind stress directly, as a forcing.

The code distinguishes between open (or cyclic) boundaries, which can allow movement across the boundaries if there is no land in the way, and the land mask, which is used to close the boundaries.  For the box tests, I usually specify open boundaries and then define the land mask as desired.  For the 2001 case we want land cells all the way around the domain.  For other tests, e.g. exercising the restoring options, we'll want some of the boundary cells to not be land.  Does that make sense?

We may want to configure these tests for more than one processor, eventually.  I guess that doesn't work because the analytical forcing functions and initial state should be defined globally, and then they need to be broadcast to the processors and blocks.

rallard77 commented:
Elizabeth,

Thanks. I will look more closely at the edeco code your previously 
provided...

Rick

On 8/8/2018 12:55 PM, Elizabeth Hunke wrote:
>
> kdyn=0 defeats the purpose of the 2001 test -- it turns the dynamics 
> off and leaves the thermo on. We need to turn the thermo off and have 
> the dynamics on. Turning the thermo off is a little complicated -- 
> there is not a namelist flag to do that. It will need to be added. 
> However, the wind stress needed for the dynamics is calculated as part 
> of the thermodynamics. In the past I've probably handled that by 
> specifying the wind stress directly, as a forcing.
>
> The code distinguishes between open (or cyclic) boundaries, which can 
> allow movement across the boundaries if there is no land in the way, 
> and the land mask, which is used to close the boundaries. For the box 
> tests, I usually specify open boundaries and then define the land mask 
> as desired. For the 2001 case we want land cells all the way around 
> the domain. For other tests, e.g. exercising the restoring options, 
> we'll want some of the boundary cells to not be land. Does that make 
> sense?
>
> We may want to configure these tests for more than one processor, 
> eventually. I guess that doesn't work because the analytical forcing 
> functions and initial state should be defined globally, and then they 
> need to be broadcast to the processors and blocks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-411495819>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h6QasqoYFWKKceRn-KXjuSA2g8djks5uOyYbgaJpZM4Uvuf4>.
>

duvivier commented:
@mattdturner 
The location of the documentation to which you will need to add the namelist info is:
/doc/source/user_guide/ug_case_settings.rst
Search the file for the table of namelist options. Add dxrect and dyrect to the grid_nml section and coriolis to the domain_nml section. 

I assume there will also be some more detailed documentation as well about the box model test and how to implement it. I think this should go in this file: 
/doc/source/user_guide/ug_testing.rst

When you get these files started I can help with making sure the references are done correctly, etc. We already reference the Hunke 2001 paper, so we will just need to make sure you use the correct reference when describing the test.

mattdturner commented:
Code has been updated per recent comments.  It now needs to be tested to confirm correct results.  I have tested the `box2001` test case, and it runs without error.  I have also run the entire `base_suite` using the `intel` compiler, and all tests pass.  So these edits do not appear to break any currently tested cases.  But I do not have any test to confirm correct results for the `box2001` case.

The documentation still needs to be updated.

mattdturner commented:
Documentation has been updated.  A draft of the ``box2001`` test overview can be seen [here](https://mattdturner-cice.readthedocs.io/en/box2001/user_guide/ug_testing.html#specific-test-cases).  A draft of the updates to the namelist variable can be seen [here](https://mattdturner-cice.readthedocs.io/en/box2001/user_guide/ug_case_settings.html).

The namelist modifications include new namelist options:

- ``dxrect`` and ``dyrect`` in ``grid_nml``
- ``thermo`` and ``coriolis`` in ``dynamics_nml``

rallard77 commented:
Tony,

You raise several good points. We can clean up the code modifications. 
I'll look at your proposed option for setting kterm = -1, as well the 
other issues related to adding the land boundary, forcing etc. I'll be 
back in the office Monday.

Thanks,
Rick

On 9/12/2018 4:24 PM, Tony Craig wrote:
>
> *@apcraig* requested changes on this pull request.
>
> I am reviewing for SE, not science.
>
> Please update indentation in ice_dyn_shared.F90
>
>   * line 34 needs to be shifted left
>   * indents around coriolis if block line 137-143
>
> Do we want to keep the write statement at line 166 in ice_dyn_shared? 
> Or should we comment that out by default. That looks like a "debug" 
> output.
>
> Wondering about the new "thermo" variable. Should we make this a 
> logical instead of an integer. It seems to serve no other purpose than 
> to turn the thermo on or off. Alternatively, should we use ktherm and 
> make -1 be an "off" option and drop thermo all together?
>
> Could we indent lines 210-231 for the new if test in CICE_RunMod.F90.
>
> The mods in cice/CICE_RunMod.F90 may need to be implemented in other 
> CICE_RunMod.F90 routines as well if we want to support the "thermo" 
> feature for other coupled models.
>
> Line 220 of CICE_RunMod.F90 suggests that if thermo is off, then we 
> are doing the box problem and will impose some forcing. I'm not sure 
> that's robust logic. First, I'm not sure the forcing should be defined 
> in this part of the code. And second, it should maybe be associated 
> with a forcing option, not with whether the thermo is on or off.
>
> The if test on line 1194 in ice_grid.F90 looks a little dicey. Could 
> this code be hit if the gridsize is 80x80 but it's not the gbox80 
> case? Are there possible cases where we want to set land on the edges? 
> I would almost recommend adding a new namelist, something like 
> "land_override" that would be off by default and on for the gbox80 
> case and it would force the land cells at 1,2, nx_global-1, nx_global 
> (and same for j) to be land without the hardwired 79,80. Use the 
> global sizes there so it's more flexible. Maybe land_override could be 
> an integer with 0=do_nothing, 1=land_on_edges, and that leaves a 
> chance for more options later.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#pullrequestreview-154850646>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h3gWVmFTgY81gsfiKW192Rbtl_kvks5uaXuggaJpZM4Uvuf4>.
>

-- 

logo

*Richard A. Allard *
*Head, Nearshore and Coupled Models Systems Section*
*Oceanography Division NRL Code 7322*
*Stennis Space Center, MS 39529*
*T: 228.688.4894*
richard.allard@nrlssc.navy.mil <mailto:richard.allard@nrlssc.navy.mil>
****

******

mattdturner commented:
> Please update indentation in ice_dyn_shared.F90
>
> line 34 needs to be shifted left
> indents around coriolis if block line 137-143
>
> Do we want to keep the write statement at line 166 in ice_dyn_shared? Or should we comment that out by default. That looks like a "debug" output.
>
> ...
>
> Could we indent lines 210-231 for the new if test in CICE_RunMod.F90.

I've updated the indentation, and removed the "debug" output in ice_dyn_shared.  I'll leave it to @rallard77 to address the other comments.

eclare108213 commented:
This is a lot closer.  The ice area and thickness are still changing, though, because transport and ridging are probably still on.  The 2001 test was *only* for calculating ice velocity and the internal stresses -- everything else is fixed.  I loosened that restriction when we started testing the elastic-decohesive model (which didn't make it into the main code). To make these box cases more versatile, I think we need to add a namelist variable to turn off each capability:  thermodynamics, dynamics (meaning the momentum and stress equations solved in evp/eap), transport and ridging.  Two have existing flags:  ktherm, kdyn.  We could create ktransport and kridge.  (Trivia:  the 'k' prefix came from POP's naming convention and stands for "kind".)  I recommend using -1 to mean 'off'.  kdyn=0 means off and we can keep that for backwards compatibility, and also add -1 as an option for it.

Also, we're moving away from using 'default' as a namelist value, so for the coriolis option, how about 'latitude' instead of 'default'?

Finally, rather than using 'land-override', I'm wondering if it makes sense to use the existing ew_boundary_type and nw_boundary_type, adding a 'land' value for them which would likely only be used for the box problems.  I disabled the 'closed' option for those because I could never get the corners to work properly in the boundary condition routines (and I think 'closed' might still be implemented there, since that code originally came from POP and I tried not to change it too much).

rallard77 commented:
Thanks for the feedback and suggestions. I'll work on making these changes.

Rick

On 9/30/2018 8:13 PM, Elizabeth Hunke wrote:
>
> This is a lot closer. The ice area and thickness are still changing, 
> though, because transport and ridging are probably still on. The 2001 
> test was /only/ for calculating ice velocity and the internal stresses 
> -- everything else is fixed. I loosened that restriction when we 
> started testing the elastic-decohesive model (which didn't make it 
> into the main code). To make these box cases more versatile, I think 
> we need to add a namelist variable to turn off each capability: 
> thermodynamics, dynamics (meaning the momentum and stress equations 
> solved in evp/eap), transport and ridging. Two have existing flags: 
> ktherm, kdyn. We could create ktransport and kridge. (Trivia: the 'k' 
> prefix came from POP's naming convention and stands for "kind".) I 
> recommend using -1 to mean 'off'. kdyn=0 means off and we can keep 
> that for backwards compatibility, and also add -1 as an option for it.
>
> Also, we're moving away from using 'default' as a namelist value, so 
> for the coriolis option, how about 'latitude' instead of 'default'?
>
> Finally, rather than using 'land-override', I'm wondering if it makes 
> sense to use the existing ew_boundary_type and nw_boundary_type, 
> adding a 'land' value for them which would likely only be used for the 
> box problems. I disabled the 'closed' option for those because I could 
> never get the corners to work properly in the boundary condition 
> routines (and I think 'closed' might still be implemented there, since 
> that code originally came from POP and I tried not to change it too much).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-425766971>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h2eE_7YHrr4hc0F59vCWWN1Z8EmQks5ugWw-gaJpZM4Uvuf4>.
>

mattdturner commented:
I've resolved merge conflicts.  I am currently running tests to ensure that nothing is broken.  I'll update once the testing is complete.

mattdturner commented:
After fixing some errors introduced during the merge conflict resolution, everything appears to be working properly now.  After running the `base_suite`:
```
90 of 109 tests PASSED
1 of 109 tests FAILED
3 of 109 tests PENDING
```
The failing test is 
```
COPY conrad_intel_restart_gbox128_4x4_boxrestore build
PASS conrad_intel_restart_gbox128_4x4_boxrestore initialrun
PASS conrad_intel_restart_gbox128_4x4_boxrestore run -1 -1 -1
FAIL conrad_intel_restart_gbox128_4x4_boxrestore test
```

eclare108213 commented:
@mattdturner, is the code throwing an error for this remaining failing test, or is it not bfb?  I'm guessing it's the latter, but I'm not sure what the three -1s mean in the 'run' line.  Are those timings?

mattdturner commented:
The failure was a result of non-bfb data.  Comparing the last output for the log files shows:
```
istep1:       240    idate:  19970111    sec:         0
                               Base Run                     Restart Run
total ice area  (km^2) = 7.14239997662472259E+06     7.14239997664026916E+06
total ice volume (k^3) = 1.42847999964811172E+13     1.42847999964819570E+13
```

There are other differences as well, but I just picked out 2.

Regarding the `-1`s, it appears to be a result of the fact that the log file is full of `*`s.  When using `grep` to get the timers, it returns `Binary file logs/cice.runlog.181012-140503 matches` instead of showing the actual timer.  As a result, the timers are filled with `-1`.  The `-1`s appears to be the value for every case in the `base_suite`.  

I will look into what is causing the asterisks to be in the diagnostic output.

eclare108213 commented:
Is the extended-grid restart capability turned on?  If not, try that and see if it fixes the nonBFB issue.  It's restart_ext or something like that.  The asterisks are probably caused by the point locations not being in the bounds of the box latitudes/longitudes.  One way to fix that would be to choose different lat/lon values for print_points.  However, if the asterisks are for global averages, then one of the hemispheres is probably missing.  Since the lat/lon values don't matter for these problems (as long as Coriolis is turned off), you could define the top half of the box to be NH latitudes and the bottom half as SH.  Just a suggestion, not sure if that would actually work.

mattdturner commented:
There asterisks were not the problem.  I had introduced a bug in the namelist writing portion of `ice_init.F90` that caused binary data to be written to the log for `kridge` and `ktransport`.  That is what was causing the problem.  After fixing that bug, 
```
COPY conrad_intel_restart_gbox128_4x4_boxrestore build
PASS conrad_intel_restart_gbox128_4x4_boxrestore initialrun
PASS conrad_intel_restart_gbox128_4x4_boxrestore run 65.69 27.16 4.60
FAIL conrad_intel_restart_gbox128_4x4_boxrestore test 

100 of 116 tests PASSED
1 of 116 tests FAILED
0 of 116 tests PENDING
```
`restart_ext` is enabled for the failing test.

eclare108213 commented:
The failing case *might* be because the wind forcing is time dependent.  It's something to check, at least.  If that is the problem, we could make the wind is constant in time -- I'm not sure that complexity is necessary here, although it would be nice to compare with the published results.

mattdturner commented:
>   I think I added that for some reason a while back, but it is not necessary. We will remove it.

Correct, it was used at one point on this branch but is no longer used.  I just pushed a commit to remove it.

Regarding the `boxrestore` case, I do believe that it is PASSing in master.  Unfortunately, I am not familiar enough with the CICE code or the modifications in this branch to quickly diagnose the problem.  Nonetheless, I'll look into it.

eclare108213 commented:
I found the problem causing the case to fail:  when the thermodynamics was turned off (setting ktherm=-1 in the box tests), aice_init and the other initial state arrays weren't getting set.  This didn't show up unless the initial ocean velocity was nonzero, which caused the initial ice velocity to be set differently when restarting.  I also cleaned up a number of other things in the code.  

I haven't run a full test suite yet.  

Still to do:
 - change coriolis = 'default' in ice_init.F90 to 'latitude'
 - land_override is non-intuitive to me.  It is essentially a land mask (usually 0=land, 1=ocean), but here 0 means the land mask is *not* applied.  It would make more sense to me to call it close_boundaries instead of land_override and make it a logical variable instead of 0 or 1.

eclare108213 commented:
@mattdturner @apcraig 
Please, would one of you run this through the full test suite?  It looks like it's pretty close now, but I'm not sure if I messed up other cases with what I did.  I'll work on bgc stuff for a while today with @njeffery 's help, and work on this PR later if necessary.   I want to check where things stand for an advection box test and stand-still testing, etc, but that doesn't need to be done as part of this PR.

mattdturner commented:
I have the `base_suite` running with the `intel` compiler.  Jobs are currently sitting in the queue, but I will update with the results once the suite completes.

mattdturner commented:
```
97 of 114 tests PASSED
0 of 114 tests FAILED
1 of 114 tests PENDING
```
The PEND test is 
```
COPY conrad_intel_restart_gx1_40x4_droundrobin_short build
RUN  conrad_intel_restart_gx1_40x4_droundrobin_short run 
PEND conrad_intel_restart_gx1_40x4_droundrobin_short test 
PASS conrad_intel_restart_gx1_40x4_droundrobin_short initialrun
```
It ran out of wallclock time.

So it looks as if everything is PASSing.

eclare108213 commented:
97+1=98 .... what happened to the other 114-98=6 tests?


Sent with BlackBerry Work
(www.blackberry.com)

From: Matthew Turner <notifications@github.com<mailto:notifications@github.com>>
Date: Wednesday, Oct 17, 2018, 12:43 PM
To: CICE-Consortium/CICE <CICE@noreply.github.com<mailto:CICE@noreply.github.com>>
Cc: Hunke, Elizabeth Clare <eclare@lanl.gov<mailto:eclare@lanl.gov>>, Review requested <review_requested@noreply.github.com<mailto:review_requested@noreply.github.com>>
Subject: Re: [CICE-Consortium/CICE] Implement box model test from 2001 JCP paper (#151)


97 of 114 tests PASSED 0 of 114 tests FAILED 1 of 114 tests PENDING

The PEND test is

COPY conrad_intel_restart_gx1_40x4_droundrobin_short build
RUN  conrad_intel_restart_gx1_40x4_droundrobin_short run
PEND conrad_intel_restart_gx1_40x4_droundrobin_short test
PASS conrad_intel_restart_gx1_40x4_droundrobin_short initialrun


It ran out of wallclock time.

So it looks as if everything is PASSing.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-430702300>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHy5n41OQA5ZzNAfpwMoDtb-btx-xo9Sks5ul15CgaJpZM4Uvuf4>.

duvivier commented:
@eclare108213 I've been confused with this total numbers of tests as well for exactly what you describe. I *think* with the dynamic allocation now in the code we don't build every single test directly. Some build and then the executable is copied over because the "missing" tests say "COPY" in the results.log file. I've been meaning to ask @apcraig about this and how we account for the tests because I also found it confusing.

rallard77 commented:
Sure, I will make those changes.

Rick

On 10/17/2018 3:46 PM, Elizabeth Hunke wrote:
>
> *@eclare108213* requested changes on this pull request.
>
> Let's change land_override to an more intuitive logical flag, e.g. 
> close_boundaries.
> The places in the documentation that reference 'thermo' here need to 
> be changed to use ktherm instead (with -1 instead of 0 as 'off').
> Instead of 'default' in ice_init.F90, set coriolis='latitude'.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#pullrequestreview-165827884>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h6_IdNeU6jeWqBKTfJq5T0itLrCIks5ul5cJgaJpZM4Uvuf4>.
>

apcraig commented:
Thanks for fixing the test reporting.  I've been aware of that issue, but I really only care about fails, not the other info.  But it makes sense for it to be consistent.

apcraig commented:
Do you want me to run a full test suite on conrad with all 4 compilers?  Or are we ok at this point?

eclare108213 commented:
I think we're okay for now.  Let's let @rallard77 or @mattdturner finish up my last few requests, then do a thorough set of tests.

mattdturner commented:
`land_override` has been replaced with `close_boundaries`.  For Coriolis, `default` has been replaced with `latitude`.    I have also updated the documentation to include `kridge` and `ktransport`.

There were merge conflicts, and those have been resolved.  I am currently running the `base_suite` to ensure that nothing is broken.

eclare108213 commented:
I think this one is ready to go now.  I still need to check the output carefully, but it appears to be doing the right things.   @apcraig, please review

eclare108213 commented:
The wind stress is zero -- I'm doing a little debugging...

eclare108213 commented:
I think this PR is ready to be merged.   I'm waiting for 2 tests to complete in the base_suite then will post a link to those results.  Here are plots from the output at day 10, a selection of which I suggest we post on the wiki as part of the release, on a "sample output" page as we have in Icepack.

<img width="818" alt="ocncurrent" src="https://user-images.githubusercontent.com/8173983/47269783-d770da80-d51f-11e8-8054-fddfa9b32fb6.png">
<img width="819" alt="wind" src="https://user-images.githubusercontent.com/8173983/47269787-d8097100-d51f-11e8-812d-1e333359fcf8.png">
<img width="818" alt="strair" src="https://user-images.githubusercontent.com/8173983/47269784-d770da80-d51f-11e8-9b27-e44021da1c3a.png">
<img width="817" alt="icevec" src="https://user-images.githubusercontent.com/8173983/47269782-d770da80-d51f-11e8-9d3b-04ad7300a497.png">
<img width="820" alt="uvel" src="https://user-images.githubusercontent.com/8173983/47269785-d770da80-d51f-11e8-8acd-e716dbee3822.png">
<img width="819" alt="vvel" src="https://user-images.githubusercontent.com/8173983/47269786-d770da80-d51f-11e8-9e53-79f1b7f18d97.png">
<img width="819" alt="divu" src="https://user-images.githubusercontent.com/8173983/47269780-d6d84400-d51f-11e8-887d-5c8677b0d308.png">
<img width="645" alt="ellipse" src="https://user-images.githubusercontent.com/8173983/47269781-d770da80-d51f-11e8-8331-a80b51f6c22c.png">

eclare108213 commented:
base_suite results:
https://github.com/CICE-Consortium/Test-Results/wiki/2689ed4109.pinto.intel.181021.130806

The 2 failing tests are because pinto does not have 40 processors available for those tests.

mattdturner commented:
>  I assume it's up to date, more or less, with master and the dynamic allocation implementation?

I merged `master` into this branch on October 19.  It includes the Fri Sep 28 commit of `Alloc dyn4` (#194).  So it should be up to date.

rallard77 commented:
Tony,

How does the timing information from the end of the run compare between 
the two sets of runs?

Rick

On 10/22/18 10:25 AM, Tony Craig wrote:
>
> I ran a full test suite on conrad last night, hash 4626b06 
> <https://github.com/CICE-Consortium/CICE/commit/4626b0622465347bc5f267226e70135e376e622c> 
> https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks. 
> Compared against the current master test suite run over the weekend. 
> One thing, it looks like this branch is one PR behind master. Anyway, 
> I think the results are probably OK. All the box results are 
> different. I assume that's expected. The 1 degree test did not 
> complete because it still has the incorrect namelist. This is fixed on 
> the master with the latest PR. The alt01 and alt03 tests are 
> different, although it's not clear exactly why. This is a little 
> concerning. Is that expected. The results are very different. Looking 
> at the log files, the first diagnostic is after 24 timesteps and it's 
> very different. The ice area is different by a factor of 10 (less), 
> the water balance has a bunch of terms zeroed out that weren't before. 
> It looks like something important has changed. I don't know why it's 
> just alt01 and alt03 that are affected. Is this something we need to 
> understand.
>
> In the diff below, it's (diff "new" "baseline")to < represents the new 
> results.
>
> |> 
> /p/home/apcraig/cice-consortium/cice.matt.box2001/testsuite.bx01/conrad_intel_restart_gx3_6x2_alt01.bx01>diff 
> logs/cice.runlog.181022-001223 
> $WORKDIR/CICE_BASELINES_MASTER/cice.4671a1cb60.181021-000127/conrad_intel_restart_gx3_6x2_alt01/cice.runlog.181021-003247 
> 62d61 < close_boundaries = F 68,70d66 < coriolis = latitude < kridge = 
> 1 < ktransport = 1 111c107 < bgc_data_dir = /uknown_bgc_data_dir --- > 
> bgc_data_dir = unknown_bgc_data_dir 322,325c318,321 < total ice area 
> (km^2) = 1.11487362338905167E+07 2.17446721276171021E+07 < total ice 
> extent(km^2) = 1.11487362338912636E+07 2.17446721276185624E+07 < total 
> ice volume (m^3) = 2.86493476407170586E+13 5.58781424233335000E+13 < 
> total snw volume (m^3) = 2.14060204122054199E+12 
> 4.17506420149698242E+12 --- > total ice area (km^2) = 
> 1.47407298426054806E+08 1.92905867664821625E+08 > total ice 
> extent(km^2) = 1.56324900931266248E+08 2.04107714224489629E+08 > total 
> ice volume (m^3) = 4.03222649989284062E+13 7.05731516956805625E+13 > 
> total snw volume (m^3) = 2.14201601881120215E+12 
> 4.17782204583553711E+12 328,329c324,325 < average albedo = 
> 0.81000000000000005 0.81000000000000028 < max ice volume (m) = 
> 2.56973947895774435 2.56973947895774435 --- > average albedo = 
> 0.70831954047609880 0.71239938402596259 > max ice volume (m) = 
> 2.57388329449594178 2.57388329449594178 335,342c331,338 < arwt evap 
> h2o kg in dt = 0.00000000000000000E+00 0.00000000000000000E+00 < arwt 
> frzl h2o kg in dt = 0.00000000000000000E+00 0.00000000000000000E+00 < 
> arwt frsh h2o kg in dt = 0.00000000000000000E+00 
> 0.00000000000000000E+00 < arwt ice mass (kg) = 2.62714517865375440E+16 
> 5.12402566021968160E+16 < arwt snw mass (kg) = 7.06398673602778875E+14 
> 1.37777118649400425E+15 < arwt tot mass (kg) = 2.69778504601403240E+16 
> 5.26180277886908240E+16 < arwt tot mass chng(kg) = 
> 0.00000000000000000E+00 0.00000000000000000E+00 < arwt water flux = 
> 0.00000000000000000E+00 0.00000000000000000E+00 --- > arwt evap h2o kg 
> in dt = -3.14065194974866943E+12 -3.93173023171445703E+12 > arwt frzl 
> h2o kg in dt = 5.27300119967046328E+13 6.62368055911166094E+13 > arwt 
> frsh h2o kg in dt = -2.64696243743919656E+14 -3.33722669005428875E+14 
> > arwt ice mass (kg) = 3.69755170040173520E+16 6.47155801049390720E+16 
> > arwt snw mass (kg) = 7.06865286207696750E+14 1.37868127512572725E+15 
> > arwt tot mass (kg) = 3.76823822902250480E+16 6.60942613800648000E+16 
> > arwt tot mass chng(kg) = 3.14285603790864000E+14 
> 3.96027744364800000E+14 > arwt water flux = 3.14285603790875625E+14 
> 3.96027744364831062E+14 344c340 < water flux error = 
> 0.00000000000000000E+00 0.00000000000000000E+00 --- > water flux error 
> = 3.08499603620219322E-16 4.69972723068647549E-16 346,352c342,348 < 
> arwt atm heat flux (W) = 2.00677252210029300E+15 
> 3.91404098297107500E+15 < arwt ocn heat flux (W) = 
> 0.00000000000000000E+00 0.00000000000000000E+00 < arwt frzl heat 
> flux(W) = 7.05454050741469920E+16 8.86156119504524640E+16 < arwt tot 
> energy (J) = -9.01060205368686569E+21 -1.75744212814227298E+22 < arwt 
> net heat (J) = -2.46739077187368092E+20 -3.04925655482932986E+20 < 
> arwt tot energy chng(J)= 0.00000000000000000E+00 
> 0.00000000000000000E+00 < arwt heat error = 2.73831954532283359E-02 
> 1.73505374999322788E-02 --- > arwt atm heat flux (W) = 
> -2.42665465720147080E+16 -3.05972704306722960E+16 > arwt ocn heat flux 
> (W) = 2.57468797943184084E-21 5.02171230614227945E-21 > arwt frzl heat 
> flux(W) = 4.89217333524982200E+15 6.14530362984248700E+15 > arwt tot 
> energy (J) = -1.25859156849351658E+22 -2.20754833009416549E+22 > arwt 
> net heat (J) = -1.04971391666152309E+20 -1.32273266617853215E+20 > 
> arwt tot energy chng(J)= -1.04971391666151752E+20 
> -1.32273266617942540E+20 > arwt heat error = 4.42602679014267966E-17 
> -4.04636975699597543E-15 359,362c355,358 < arwt salt mass (kg) = 
> 1.05085807146150172E+14 2.04961026408787281E+14 < arwt salt mass 
> chng(kg)= 0.00000000000000000E+00 0.00000000000000000E+00 < arwt salt 
> flx in dt(kg)= 0.00000000000000000E+00 0.00000000000000000E+00 < arwt 
> salt flx error = 0.00000000000000000E+00 0.00000000000000000E+00 --- > 
> arwt salt mass (kg) = 1.47902068016069406E+14 2.58862320419756281E+14 
> > arwt salt mass chng(kg)= 1.04614516461391748E+12 
> 1.31901317799086157E+12 > arwt salt flx in dt(kg)= 
> -1.04614516461395789E+12 -1.31901317799098901E+12 > arwt salt flx 
> error = -2.73189374425175273E-16 -4.92313466260164250E-16 
> 376,377c372,373 < avg ice thickness (m) = 2.56973947895791621 
> 2.56973947895791621 < avg snow depth (m) = 0.19200400801603207 
> 0.19200400801603207 --- > avg ice thickness (m) = 2.57388329449611364 
> 2.57388329449611364 > avg snow depth (m) = 0.19213083652475232 
> 0.19213083652475232 380c376 < surface temperature(C) = 
> -20.14999999999998082 -20.14999999999998082 --- > surface 
> temperature(C) = -24.57951004561716246 -24.57951004561716246 
> 382,385c378,381 < outward longwave flx = 0.00000000000000000 
> 0.00000000000000000 < sensible heat flx = 0.00000000000000000 
> 0.00000000000000000 < latent heat flx = 0.00000000000000000 
> 0.00000000000000000 < subl/cond (m ice) = 0.00000000000000000 
> 0.00000000000000000 --- > outward longwave flx = 
> -214.66891827616245791 -214.66891827616245791 > sensible heat flx = 
> 18.62840040205053782 18.62840040205053782 > latent heat flx = 
> 1.36331636320478089 1.36331636320478089 > subl/cond (m ice) = 
> 0.00000188789027464 0.00000188789027464 389,390c385,386 < new ice (m) 
> = 0.00000000000000000 0.00000000000000000 < congelation (m) = 
> 0.00000000000000000 0.00000000000000000 --- > new ice (m) = 
> 0.00000000000000038 0.00000000000000038 > congelation (m) = 
> 0.00017251622852200 0.00017251622852200 393,395c389,391 < effective 
> dhi (m) = 0.00000000000000000 0.00000000000000000 < effective dhs (m) 
> = 0.00000000000000000 0.00000000000000000 < intnl enrgy chng(W/m^2)= 
> 0.00000000000000000 0.00000000000000000 --- > effective dhi (m) = 
> 0.00017251622852221 0.00017251622852221 > effective dhs (m) = 
> 0.00000524604661165 0.00000524604661165 > intnl enrgy chng(W/m^2)= 
> 14.83781797137525338 14.83781797137525338 400c396 < heat used (W/m^2) 
> = 0.00000000000000000 0.00000000000000000 --- > heat used (W/m^2) = 
> -0.00000000000000000 -0.00000000000000000 |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-431868157>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h3Db0KTVxokLzuu4UCfYFP_PBIS7ks5uneOCgaJpZM4Uvuf4>.
>

rallard77 commented:
Tony,

How does the timing information from the end of the run compare between 
the two sets of runs?

Rick

On 10/22/2018 10:25 AM, Tony Craig wrote:
>
> I ran a full test suite on conrad last night, hash 4626b06 
> <https://github.com/CICE-Consortium/CICE/commit/4626b0622465347bc5f267226e70135e376e622c> 
> https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks. 
> Compared against the current master test suite run over the weekend. 
> One thing, it looks like this branch is one PR behind master. Anyway, 
> I think the results are probably OK. All the box results are 
> different. I assume that's expected. The 1 degree test did not 
> complete because it still has the incorrect namelist. This is fixed on 
> the master with the latest PR. The alt01 and alt03 tests are 
> different, although it's not clear exactly why. This is a little 
> concerning. Is that expected. The results are very different. Looking 
> at the log files, the first diagnostic is after 24 timesteps and it's 
> very different. The ice area is different by a factor of 10 (less), 
> the water balance has a bunch of terms zeroed out that weren't before. 
> It looks like something important has changed. I don't know why it's 
> just alt01 and alt03 that are affected. Is this something we need to 
> understand.
>
> In the diff below, it's (diff "new" "baseline")to < represents the new 
> results.
>
> |> 
> /p/home/apcraig/cice-consortium/cice.matt.box2001/testsuite.bx01/conrad_intel_restart_gx3_6x2_alt01.bx01>diff 
> logs/cice.runlog.181022-001223 
> $WORKDIR/CICE_BASELINES_MASTER/cice.4671a1cb60.181021-000127/conrad_intel_restart_gx3_6x2_alt01/cice.runlog.181021-003247 
> 62d61 < close_boundaries = F 68,70d66 < coriolis = latitude < kridge = 
> 1 < ktransport = 1 111c107 < bgc_data_dir = /uknown_bgc_data_dir --- > 
> bgc_data_dir = unknown_bgc_data_dir 322,325c318,321 < total ice area 
> (km^2) = 1.11487362338905167E+07 2.17446721276171021E+07 < total ice 
> extent(km^2) = 1.11487362338912636E+07 2.17446721276185624E+07 < total 
> ice volume (m^3) = 2.86493476407170586E+13 5.58781424233335000E+13 < 
> total snw volume (m^3) = 2.14060204122054199E+12 
> 4.17506420149698242E+12 --- > total ice area (km^2) = 
> 1.47407298426054806E+08 1.92905867664821625E+08 > total ice 
> extent(km^2) = 1.56324900931266248E+08 2.04107714224489629E+08 > total 
> ice volume (m^3) = 4.03222649989284062E+13 7.05731516956805625E+13 > 
> total snw volume (m^3) = 2.14201601881120215E+12 
> 4.17782204583553711E+12 328,329c324,325 < average albedo = 
> 0.81000000000000005 0.81000000000000028 < max ice volume (m) = 
> 2.56973947895774435 2.56973947895774435 --- > average albedo = 
> 0.70831954047609880 0.71239938402596259 > max ice volume (m) = 
> 2.57388329449594178 2.57388329449594178 335,342c331,338 < arwt evap 
> h2o kg in dt = 0.00000000000000000E+00 0.00000000000000000E+00 < arwt 
> frzl h2o kg in dt = 0.00000000000000000E+00 0.00000000000000000E+00 < 
> arwt frsh h2o kg in dt = 0.00000000000000000E+00 
> 0.00000000000000000E+00 < arwt ice mass (kg) = 2.62714517865375440E+16 
> 5.12402566021968160E+16 < arwt snw mass (kg) = 7.06398673602778875E+14 
> 1.37777118649400425E+15 < arwt tot mass (kg) = 2.69778504601403240E+16 
> 5.26180277886908240E+16 < arwt tot mass chng(kg) = 
> 0.00000000000000000E+00 0.00000000000000000E+00 < arwt water flux = 
> 0.00000000000000000E+00 0.00000000000000000E+00 --- > arwt evap h2o kg 
> in dt = -3.14065194974866943E+12 -3.93173023171445703E+12 > arwt frzl 
> h2o kg in dt = 5.27300119967046328E+13 6.62368055911166094E+13 > arwt 
> frsh h2o kg in dt = -2.64696243743919656E+14 -3.33722669005428875E+14 
> > arwt ice mass (kg) = 3.69755170040173520E+16 6.47155801049390720E+16 
> > arwt snw mass (kg) = 7.06865286207696750E+14 1.37868127512572725E+15 
> > arwt tot mass (kg) = 3.76823822902250480E+16 6.60942613800648000E+16 
> > arwt tot mass chng(kg) = 3.14285603790864000E+14 
> 3.96027744364800000E+14 > arwt water flux = 3.14285603790875625E+14 
> 3.96027744364831062E+14 344c340 < water flux error = 
> 0.00000000000000000E+00 0.00000000000000000E+00 --- > water flux error 
> = 3.08499603620219322E-16 4.69972723068647549E-16 346,352c342,348 < 
> arwt atm heat flux (W) = 2.00677252210029300E+15 
> 3.91404098297107500E+15 < arwt ocn heat flux (W) = 
> 0.00000000000000000E+00 0.00000000000000000E+00 < arwt frzl heat 
> flux(W) = 7.05454050741469920E+16 8.86156119504524640E+16 < arwt tot 
> energy (J) = -9.01060205368686569E+21 -1.75744212814227298E+22 < arwt 
> net heat (J) = -2.46739077187368092E+20 -3.04925655482932986E+20 < 
> arwt tot energy chng(J)= 0.00000000000000000E+00 
> 0.00000000000000000E+00 < arwt heat error = 2.73831954532283359E-02 
> 1.73505374999322788E-02 --- > arwt atm heat flux (W) = 
> -2.42665465720147080E+16 -3.05972704306722960E+16 > arwt ocn heat flux 
> (W) = 2.57468797943184084E-21 5.02171230614227945E-21 > arwt frzl heat 
> flux(W) = 4.89217333524982200E+15 6.14530362984248700E+15 > arwt tot 
> energy (J) = -1.25859156849351658E+22 -2.20754833009416549E+22 > arwt 
> net heat (J) = -1.04971391666152309E+20 -1.32273266617853215E+20 > 
> arwt tot energy chng(J)= -1.04971391666151752E+20 
> -1.32273266617942540E+20 > arwt heat error = 4.42602679014267966E-17 
> -4.04636975699597543E-15 359,362c355,358 < arwt salt mass (kg) = 
> 1.05085807146150172E+14 2.04961026408787281E+14 < arwt salt mass 
> chng(kg)= 0.00000000000000000E+00 0.00000000000000000E+00 < arwt salt 
> flx in dt(kg)= 0.00000000000000000E+00 0.00000000000000000E+00 < arwt 
> salt flx error = 0.00000000000000000E+00 0.00000000000000000E+00 --- > 
> arwt salt mass (kg) = 1.47902068016069406E+14 2.58862320419756281E+14 
> > arwt salt mass chng(kg)= 1.04614516461391748E+12 
> 1.31901317799086157E+12 > arwt salt flx in dt(kg)= 
> -1.04614516461395789E+12 -1.31901317799098901E+12 > arwt salt flx 
> error = -2.73189374425175273E-16 -4.92313466260164250E-16 
> 376,377c372,373 < avg ice thickness (m) = 2.56973947895791621 
> 2.56973947895791621 < avg snow depth (m) = 0.19200400801603207 
> 0.19200400801603207 --- > avg ice thickness (m) = 2.57388329449611364 
> 2.57388329449611364 > avg snow depth (m) = 0.19213083652475232 
> 0.19213083652475232 380c376 < surface temperature(C) = 
> -20.14999999999998082 -20.14999999999998082 --- > surface 
> temperature(C) = -24.57951004561716246 -24.57951004561716246 
> 382,385c378,381 < outward longwave flx = 0.00000000000000000 
> 0.00000000000000000 < sensible heat flx = 0.00000000000000000 
> 0.00000000000000000 < latent heat flx = 0.00000000000000000 
> 0.00000000000000000 < subl/cond (m ice) = 0.00000000000000000 
> 0.00000000000000000 --- > outward longwave flx = 
> -214.66891827616245791 -214.66891827616245791 > sensible heat flx = 
> 18.62840040205053782 18.62840040205053782 > latent heat flx = 
> 1.36331636320478089 1.36331636320478089 > subl/cond (m ice) = 
> 0.00000188789027464 0.00000188789027464 389,390c385,386 < new ice (m) 
> = 0.00000000000000000 0.00000000000000000 < congelation (m) = 
> 0.00000000000000000 0.00000000000000000 --- > new ice (m) = 
> 0.00000000000000038 0.00000000000000038 > congelation (m) = 
> 0.00017251622852200 0.00017251622852200 393,395c389,391 < effective 
> dhi (m) = 0.00000000000000000 0.00000000000000000 < effective dhs (m) 
> = 0.00000000000000000 0.00000000000000000 < intnl enrgy chng(W/m^2)= 
> 0.00000000000000000 0.00000000000000000 --- > effective dhi (m) = 
> 0.00017251622852221 0.00017251622852221 > effective dhs (m) = 
> 0.00000524604661165 0.00000524604661165 > intnl enrgy chng(W/m^2)= 
> 14.83781797137525338 14.83781797137525338 400c396 < heat used (W/m^2) 
> = 0.00000000000000000 0.00000000000000000 --- > heat used (W/m^2) = 
> -0.00000000000000000 -0.00000000000000000 |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CICE-Consortium/CICE/pull/151#issuecomment-431868157>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/Ab21h3Db0KTVxokLzuu4UCfYFP_PBIS7ks5uneOCgaJpZM4Uvuf4>.
>

eclare108213 commented:
I don't expect the results to be different.  Maybe for the box tests, but not other configurations.  The results are so different, my first suspicion is that the forcing has changed.  We need to figure this out before merging.

mattdturner commented:
Rick noticed that the issues appeared to be related to the `ktherm` namelist variable.  I noticed in `CICE_RunMod.F90` that we used `if (ktherm > 0) then` to disable thermodynamics.  However, a `ktherm` value of 0 does not mean thermodynamics is disabled.  I updated the conditionals to read `if (ktherm >= 0) then` so that thermodynamics is only disabled if `ktherm = -1`.  I believe that this should resolve the problem, but it should be tested.

I'm also not sure what the `CICE_RunMod.F90_debug` file is, or if it should be in the repository or not.  But I updated that file as well.

apcraig commented:
I'm running a full suite on conrad with intel to see if it's fixed.

eclare108213 commented:
That makes a lot of sense.  The tests that failed are likely the ones that have ktherm=0 set.

CICE_RunMod.F90_debug is supposed to be identical to CICE_RunMod.F90, but with the addition of calls to a debugging routine that prints tons of information.  There's probably a better way to do this without wrapping all of those calls in "if debug" conditionals.  I've kept it as a separate file because it's rarely used -- but maybe it's rarely used because it's a separate file and no one else knows what it's for...

apcraig commented:
I think we're OK now based on the conrad intel results.  I think we can merge.  thanks.

apcraig commented:
Does anyone think we should wait to merge?

eclare108213 commented:
no need to wait -- go for it

apcraig commented:
Done.  And just for fun, I'm going to run a full test suite on conrad again with master.
